datafusion/execution/src/task.rs

```diff
@@ -138,7 +138,7 @@ impl FunctionRegistry for TaskContext {
         let result = self.scalar_functions.get(name);
 
         result.cloned().ok_or_else(|| {
-            DataFusionError::Internal(format!(
+            DataFusionError::Plan(format!(
```

Comment:
prob we could have short methods
```
plan_err(text)
execution_err(text)
...
``` 
to replace wordy
```
            DataFusionError::Plan(format!(
               text
            ))
```

Reply:
It is an excellent idea. I will try to do this

Reply:
I tried to implement this idea in https://github.com/apache/arrow-datafusion/pull/6740 but I found them harder / less easy to use